src/triage/component/catwalk/storage.py

```diff
@@ -70,114 +94,184 @@ def open(self, *args, **kwargs):
 
 
 class FSStore(Store):
+    """Store an object on the local filesystem.
+    
+    Example:
+    ```
+    store = FSStore('/mnt', 'models', 'model.pkl')
+    return store.load()
+    ```
+
+    Args:
+        *pathparts: A variable length list of components of the path, to be processed in order. 
+            All components will be joined using pathlib.Path to create the final path
+                using the correct separator for the operating system. However, if you pass
+                components that already contain a separator, those separators won't be modified
+    """
+    def __init__(self, *pathparts):
+        self.path = pathlib.Path(*pathparts)
+        os.makedirs(dirname(self.path), exist_ok=True)
+
     def exists(self):
         return os.path.isfile(self.path)
 
     def delete(self):
         os.remove(self.path)
 
     def open(self, *args, **kwargs):
-        os.makedirs(dirname(self.path), exist_ok=True)
         return open(self.path, *args, **kwargs)
 
 
-class MemoryStore(Store):
-    store = None
-
-    def exists(self):
-        return self.store is not None
-
-    def delete(self):
-        self.store = None
-
-    def write(self, bytestream):
-        self.store = bytestream
-
-    def load(self):
-        return self.store
-
-    def open(self, *args, **kwargs):
-        raise ValueError('MemoryStore objects cannot be opened and closed like files'
-                         'Use write/load methods instead.')
-
+class ProjectStorage(object):
+    """Store and access files associated with a project.
+    
+    Args:
+        project_path (string): The base path for all files in the project.
+            The scheme prefix of the path will determine the storage medium.
+    """
+    def __init__(self, project_path):
+        self.project_path = project_path
+        self.storage_class = Store.factory(self.project_path).__class__
```

Comment:
Seems like we want that nice standalone function, and factored, such that you can say:

    def select_storage_class(path):
        if path.startswith('s3://'):
            return S3Store

        ...

...which you could use above, and then the other method you want extending that:

    def make_storage(path):
        storage_class = select_storage_class(path)
        return storage_class(path)

(or whatever)